common/models/user.js

```diff
@@ -1,103 +1,46 @@
 'use strict';
 // PersistedModel
+const bcrypt = require('bcrypt');
+const saltRounds = 10;
 
 module.exports = (User) => {
 
-
-  // 1st API for Login User
-
-  User.login = (userData, cb) => {
-    let { email, password } = userData;
-    let query = { where: { "email": email } }
-    User.findOne(query)
-      .then(user => {
-        if (user){
-          let { status, id } = user;
-          if (status == "ACTIVE"){
-              if(password == user.password)
-                cb(null, user)
-              else
-                cb(null, "You have entered invalid password.")          
-          } 
-          else {
-            let OTP = Math.floor(100000 + Math.random() * 900000)
-            User.update({ _id:id }, { OTP:OTP }, (err, user) => {
-              if (err)
-                cb(err)
-              else
-                cb(null, "An OTP hase been sent on you registered email address.")
-            })
-          }
-        } 
-        else {
-          cb(null, "Email id not registered.")
-        }
-      })
-      .catch(err => cb(err))
-    }
-    let login = { 
-      http: { path: '/login', verb: "post" },
-      accepts: { arg: 'data', type: "object", http: { source: "body" } },
-      returns: { arg:"data", type:"object" }
-    }
-
-  User.remoteMethod('login', login)
-
-
-
-
-
-  // 2nd API for Singup User 
-  User.signup = (userData, cb) => {
-    let query = { where: { "email": userData.email },  }
-    User.findOne(query, (err, result) => {
-      if (err)
-        cb(err)
-      else if (!result) {
-        new User(userData).save((err, result) => {
-          if (err)
-            cb(err)
-          else
-            cb(null, result)
-        })
-      } 
-      else
-        cb(null, "Email id already exits.")
-    })
-   
-  }
-  let signup = { 
-      http: { path: '/signup', verb: "post" },
-      accepts: { arg: 'data', type: "object", http: { source: "body" } },
-      returns: [
-        { arg: 'emal', type: "string" }, 
-        { arg: 'name', type: "string" },
-        { arg: 'id',   type: 'string' }
-      ]
+  // 1st API for Signup
+  User.signup = (userData, cb)=>{
+    var user = new User(userData)
```

Comment:
where you check the validation? and what are the fields that you saved in the db